lib/renderer/web-view/web-view.js

```diff
@@ -277,69 +273,74 @@ const registerWebViewElement = function () {
     'stopFindInPage',
     'downloadURL',
     'inspectServiceWorker',
-    'print',
-    'printToPDF',
     'showDefinitionForSelection',
-    'capturePage',
     'setZoomFactor',
-    'setZoomLevel',
-    'getZoomLevel',
-    'getZoomFactor'
+    'setZoomLevel'
   ]
   const nonblockMethods = [
     'insertCSS',
     'insertText',
     'send',
     'sendInputEvent',
     'setLayoutZoomLevelLimits',
-    'setVisualZoomLevelLimits'
+    'setVisualZoomLevelLimits',
+    // with callback
+    'capturePage',
+    'executeJavaScript',
+    'getZoomFactor',
+    'getZoomLevel',
+    'print',
+    'printToPDF'
   ]
 
+  const getGuestInstanceId = function (self) {
+    const internal = v8Util.getHiddenValue(self, 'internal')
+    if (!internal.guestInstanceId) {
+      throw new Error('The WebView must be attached to the DOM and the dom-ready event emitted before this method can be called.')
```

Comment:
I think its better to move this error message back into the `createBlockHandler` and print out the old error message based on the result from `getGuestInstanceId`.